features/FEATURE_BLE/ble/generic/SecurityDb.h

```diff
@@ -584,28 +584,32 @@ class SecurityDb {
         WhitelistDbCb_t cb,
         ::Gap::Whitelist_t *whitelist
     ) {
-        for (size_t i = 0; i < get_entry_count() && i < whitelist->capacity; i++) {
+        for (size_t i = 0; i < get_entry_count() && whitelist->size < whitelist->capacity; i++) {
             entry_handle_t db_handle = get_entry_handle_by_index(i);
             SecurityDistributionFlags_t* flags = get_distribution_flags(db_handle);
 
             if (!flags) {
                 continue;
             }
 
+            SecurityEntryIdentity_t* identity = read_in_entry_peer_identity(db_handle);
+            if (!identity) {
```

Comment:
Can that be checked against `NULL`?

Reply:
the check is completely equivalent and down to style, both styles are used in mbed

Reply:
I don't like this style :). But ok ;)